### ISSUE TITLE
Add End-to-End (E2E) Tests for CLI

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#34]
### Background
Currently, gomarklint has comprehensive unit tests for individual rules and parsers, but lacks end-to-end testing that validates the complete CLI workflow. E2E tests would ensure the entire application works correctly from build to execution.

### Objectives
- Test the actual compiled binary behavior (not just library functions)
- Validate CLI flag parsing and configuration file loading
- Ensure output formats (plain text and JSON) work correctly
- Prevent regressions in the complete user workflow

#### Directory Structure

```
gomarklint/
├── e2e/
│ ├── e2e_test.go # Main E2E test suite
│ ├── helper.go # Helper functions for running CLI
│ └── fixtures/ # Test markdown files
│ ├── valid.md
│ ├── heading_errors.md
│ ├── duplicate_headings.md
│ ├── blank_line_errors.md
│ └──.gomarklint.json # Test config file
```

#### Test Approach
1. Build the binary before running tests: `go build -o gomarklint-test .`
2. Execute the binary with various arguments using `exec.Command`
3. Verify exit codes, stdout/stderr output, and error messages
4. Clean up the test binary after tests complete

#### Test Scenarios

**Basic Functionality:**
- [ ] Valid markdown file → exit 0, no errors
- [ ] File with heading level errors → exit 1, expected error messages
- [ ] File with duplicate headings → exit 1, expected error messages
- [ ] File with multiple blank lines → exit 1, expected error messages

**Configuration:**
- [ ] Load and apply `.gomarklint.json` configuration
- [ ] CLI flags override config file settings
- [ ] `--min-heading` flag works correctly
- [ ] `--disable-*` flags disable specific rules

**Output Formats:**
- [ ] Default text output shows readable error messages
- [ ] `--format=json` produces valid JSON
- [ ] JSON output includes correct file paths and line numbers

**Multiple Files:**
- [ ] Process multiple files specified as arguments
- [ ] Process entire directories recursively
- [ ] Report errors from all files correctly

**Edge Cases:**
- [ ] Non-existent file → exit 1, appropriate error
- [ ] Invalid config file → exit 1, appropriate error
- [ ] Empty file → exit 0
- [ ] File with frontmatter → correctly stripped and processed

### Example Test Code Structure
```go
func TestE2E_ValidMarkdown(t *testing.T) {
    cmd := exec.Command("./gomarklint-test", "e2e/fixtures/valid.md")
    output, err := cmd.CombinedOutput()
    
    if err != nil {
        t.Errorf("expected exit 0, got error: %v\noutput: %s", err, output)
    }
}

func TestE2E_HeadingLevelErrors(t *testing.T) {
    cmd := exec.Command("./gomarklint-test", "e2e/fixtures/heading_errors.md")
    output, err := cmd.CombinedOutput()
    
    if err == nil {
        t.Error("expected exit 1, got exit 0")
    }
    
    if !strings.Contains(string(output), "heading level") {
        t.Errorf("expected heading level error in output, got: %s", output)
    }
}
```

### Benefits
- Ensures the complete application works as users expect
- Catches integration issues between components
- Validates CLI flag parsing and configuration handling
- Provides confidence for refactoring and new features

### Implementation Checklist
- [ ] Create `e2e/` directory structure
- [ ] Create test fixture markdown files with various issues
- [ ] Implement helper functions for running CLI and checking output
- [ ] Write basic E2E test cases
- [ ] Write configuration loading tests
- [ ] Write output format tests
- [ ] Write multi-file tests
- [ ] Update CI/CD to run E2E tests
- [ ] Document E2E testing approach in README

### Notes
- E2E tests should use `go test -tags=e2e` or similar to separate from unit tests
- Consider using `t.TempDir()` for isolated test environments
- Binary build time should be minimized (build once, run many tests)
